src/test/FuzzerImpl.cpp

```diff
@@ -527,73 +713,92 @@ isBadOverlayFuzzerInput(StellarMessage const& m)
+                                   "while fuzzing: transaction result {}"),
+                        txFramePtr->getResultCode(),
+                        xdr_to_string(txFramePtr->getResult()));
+        LOG_FATAL(DEFAULT_LOG, "{}", msg);
+        throw std::runtime_error(msg);
+    }
+
+    for (auto const& opFrame : txFramePtr->getOperations())
+    {
+        auto const& op = opFrame->getOperation();
+        auto const& tr = opFrame->getResult().tr();
+        auto const opType = op.body.type();
+
+        if ((opType == MANAGE_BUY_OFFER &&
+             tr.manageBuyOfferResult().success().offer.effect() ==
+                 MANAGE_OFFER_DELETED) ||
+            (opType == MANAGE_SELL_OFFER &&
+             tr.manageSellOfferResult().success().offer.effect() ==
+                 MANAGE_OFFER_DELETED))
+        {
+            auto const msg =
+                fmt::format(FMT_STRING("MANAGE_OFFER_DELETED while setting "
+                                       "up fuzzing -- operation is {}"),
+                            xdr_to_string(op));
+            LOG_FATAL(DEFAULT_LOG, "{}", msg);
+            throw std::runtime_error(msg);
+        }
+
+        storeLedgerKeyIDsForOp(opFrame);
+    }
+}
+
+// Empties "ops" as operations are applied.  If `isSetupOperation` is `false`,
 // ignores failures and might apply further transactions after failures.
-static void
-attemptToApplyOps(LedgerTxn& ltx, PublicKey const& sourceAccount,
-                  xdr::xvector<Operation>::const_iterator begin,
-                  xdr::xvector<Operation>::const_iterator end, Application& app,
-                  bool const throwIfTxFails = true)
+// If `isSetupOperation` is `true`, throws exceptions on transaction failures,
+// and creates new stored ledger key IDs for any new `LedgerEntry`s created by
+// the applied operations.  (When operations are performed by fuzz testing
+// rather than by setup, the new IDs are created when the operations are
+// unpacked.)
+void
+TransactionFuzzer::attemptToApplyOps(
+    LedgerTxn& ltx, PublicKey const& sourceAccount,
+    xdr::xvector<Operation>::const_iterator begin,
+    xdr::xvector<Operation>::const_iterator end, Application& app,
+    bool const isSetupOperation, bool const restartSeqNums)
 {
+    bool usePostInitSeqNum = restartSeqNums;
+
     for (auto beginOpsInThisTx = begin; beginOpsInThisTx != end;)
     {
         auto endOpsInThisTx =
             std::distance(beginOpsInThisTx, end) <= MAX_OPS_PER_TX
                 ? end
                 : begin + MAX_OPS_PER_TX;
-        auto txFramePtr =
-            createFuzzTransactionFrame(sourceAccount, beginOpsInThisTx,
-                                       endOpsInThisTx, app.getNetworkID());
+
+        auto txFramePtr = createFuzzTransactionFrame(
+            ltx, *this, sourceAccount, beginOpsInThisTx, endOpsInThisTx,
+            app.getNetworkID(),
+            (usePostInitSeqNum ? mPostInitSourceSeqNum
```

Comment:
Yep, that turned out to be much nicer when simply split into two functions, rather than one with a parameter.  I've done that now (and in a commit of its own).